### ISSUE TITLE
problem with the size of the log[yii2/log/FileTarget]

### ISSUE THREAD

**Issue by stu2162583** [yiisoft/yii2#19079]
### What steps will reproduce the problem?
the website has slightly more visits

### What is the expected result?
the log should be full 10Mb

### What do you get instead?
some logs are too small

![F6F94CD9-5EFC-433a-93F1-22F6A0861CF3](https://user-images.githubusercontent.com/14155947/146102222-4fadd3c5-c19c-4754-899b-49809cf52bfc.png)

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15.1
| PHP version      | 7.1
| Operating system | linux

Remark：

This file has been manually modified to the latest code, other source file have not changed

```
// vendor/yiisoft/yii2/log/FileTarget.php
if ($this->enableRotation && @filesize($this->logFile) > $this->maxFileSize * 1024) {
            @flock($fp, LOCK_UN);
            @fclose($fp);
            // Manual modification
            $this->rotateFiles();

```

Thanks！

**Comment by stu2162583** [comment]
Sorry, I learned that our system uses lvs and nas, and the nas used for file writing may be caused by the use of nas. I have reported this to the operation and maintenance personnel. After observing for a period of time, I will come back.

**Comment by zzx-ann** [comment]
This bug has been fixed in version 2.0.43

**Comment by stu2162583** [comment]
> This bug has been fixed in version 2.0.43

thanks, i guess they are not the same problem.

**Comment by stu2162583** [comment]
hello, I'm back again, after our testing, The problem is due to rotateFiles and concurrency

So I came to ask for help.

**config**
```
'log' => [
	'targets' => [
		[
			'class' => 'yii\log\FileTarget',
			'categories' => ['open'],
			'logVars' => [],
			'maxFileSize' => 100, //for better display of errors
			'maxLogFiles' => 5,
			'logFile' => '@root/runtime/logs/open.log',
		]
	],
],
```

**code**
```
$request = Yii::$app->request;
$header = $request->getHeaders();
$get = $request->get();
$body = $request->bodyParams;
$method =  $request->method;
Yii::info([
	'header' => $header,
	'get' => $get,
	'body' => $body,
	'method' => $method,
], 'open');
```

test with ab
```
ab -n1000 -c10 http://example.com/path1/path2
```

You will see that some logs are very small
------------------------------------------------------------------------

By the way, I don't think it's a good idea to attach an application log when logvar is not empty, preferably something that can be controlled.

```
'logVars' => ['_GET', '_POST'],

// will hava
... category: application ....
```
